File: demo-vault/code/hello.go
// A sample Go program demonstrating interfaces, goroutines, and error handling.

package main

import (
	"fmt"
	"sort"
	"strings"
	"sync"
)

type Priority int

const (
	Low Priority = iota
	Medium
	High
)

func (p Priority) String() string {
	switch p {
	case High:
		return "HIGH"
	case Medium:
		return "MEDIUM"
	default:
		return "LOW"
	}
}

type Task struct {
	Title    string
	Priority Priority
	Done     bool
}

func NewTask(title string, priority Priority) Task {
	return Task{Title: title, Priority: priority}
}

func (t *Task) MarkDone() {
	t.Done = true
}

func (t Task) Summary() string {
	status := " "
	if t.Done {
		status = "x"
	}
	return fmt.Sprintf("[%s] %s (%s)", status, t.Title, t.Priority)
}

type TaskManager struct {
	mu    sync.Mutex
	tasks []Task
}

func (tm *TaskManager) Add(task Task) {
	tm.mu.Lock()
	defer tm.mu.Unlock()
	tm.tasks = append(tm.tasks, task)
}

func (tm *TaskManager) Pending() []Task {
	tm.mu.Lock()
	defer tm.mu.Unlock()
	var pending []Task
	for _, t := range tm.tasks {
		if !t.Done {
			pending = append(pending, t)
		}
	}
	return pending
}

func (tm *TaskManager) Find(title string) (*Task, bool) {
	tm.mu.Lock()
	defer tm.mu.Unlock()
	for i, t := range tm.tasks {
		if strings.EqualFold(t.Title, title) {
			return &tm.tasks[i], true
		}
	}
	return nil, false
}

func (tm *TaskManager) SortByPriority() {
	tm.mu.Lock()
	defer tm.mu.Unlock()
	sort.SliceStable(tm.tasks, func(i, j int) bool {
		return tm.tasks[i].Priority > tm.tasks[j].Priority
	})
}

func main() {
	manager := TaskManager{}
	manager.Add(NewTask("Deploy to production", High))
	manager.Add(NewTask("Write unit tests", High))
	manager.Add(NewTask("Update README", Low))

	fmt.Println("Pending tasks:")
	for _, t := range manager.Pending() {
		fmt.Println(" ", t.Summary())
	}

	if task, ok := manager.Find("Write unit tests"); ok {
		task.MarkDone()
		fmt.Printf("\nMarked done: %s\n", task.Summary())
	}

	manager.SortByPriority()
	fmt.Printf("\nSorted by priority: %d tasks\n", len(manager.tasks))
}

File: demo-vault/code/hello.rb
# frozen_string_literal: true

# A simple TaskManager in Ruby demonstrating classes, blocks, and enumerable patterns

class Task
  attr_accessor :title, :priority, :completed

  def initialize(title, priority = :medium)
    @title = title
    @priority = priority
    @completed = false
  end

  def complete!
    @completed = true
  end

  def to_s
    status = @completed ? "[x]" : "[ ]"
    "#{status} [#{@priority.upcase}] #{@title}"
  end
end

class TaskManager
  include Enumerable

  def initialize
    @tasks = []
  end

  def add(title, priority = :medium)
    @tasks << Task.new(title, priority)
  end

  def each(&block)
    @tasks.each(&block)
  end

  def pending
    @tasks.reject(&:completed)
  end

  def by_priority
    order = { high: 0, medium: 1, low: 2 }
    pending.sort_by { |t| order[t.priority] || 99 }
  end

  def summary
    groups = pending.group_by(&:priority).transform_values(&:count)
    puts "Pending tasks by priority:"
    groups.each { |pri, count| puts "  #{pri.upcase}: #{count}" }
  end
end

if __FILE__ == $PROGRAM_NAME
  tm = TaskManager.new
  tm.add("Write unit tests", :high)
  tm.add("Update README", :medium)
  tm.add("Fix typos in docs", :low)
  tm.add("Refactor auth module", :high)
  tm.find { |t| t.title == "Update README" }&.complete!

  tm.summary

  puts "\nSorted pending:"
  tm.by_priority.each { |t| puts "  #{t}" }
end
